### PR TITLE
#12 GET /api/articles/:article_id

### PR DESCRIPTION
**Pull request by almcw**
Responds with:

an article object, which should have the following properties:

author which is the username from the users table
title
article_id
body
topic
created_at
votes

Also tests for an article_id which doesn't exist and responds with 404 status and a message indicating article cannot be found.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -41,3 +41,31 @@ describe("GET /api/topics", () => {
...
+
+  test("returns status 404 with non-existant article_id", () => {
+    return request(app)
+      .get("/api/articles/199")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("No article found for article_id: 199");
+      });
+  });
+});
```
Comment (jadelandeg) [inline __tests__/app.test.js:71]: nice error testing here. Just 1 more to consider, what happens if there is a bad request with an invalid article id (incorrect data type)? Think PSQL errors/400

File: models/news.js [hunk 2]
```diff
@@ -3,3 +3,22 @@ const db = require("../db/connection");
...
+
+exports.selectArticleById = (article_id) => {
+  return db
+    .query(
+      "select users.username AS author, title, article_id, body, topic, created_at, votes from articles INNER JOIN users ON articles.author = users.username where article_id = $1;",
+      [article_id]
+    )
+    .then((result) => {
+      const article = result.rows[0];
+      console.log(article);
```
Comment (jadelandeg) [inline models/news.js:15]: get rid of this console.log if it isn't necessary anymore

**Review by jadelandeg — CHANGES_REQUESTED** [review]
Well done! Just need another bit of error handling. See comments

File: __tests__/app.test.js (hunk 1 above)
Comment (almcw) [reply __tests__/app.test.js:71]: added error handling for 22P02 PSQL error and additional test

File: models/news.js (hunk 2 above)
Comment (almcw) [reply models/news.js]: removed

**Comment by almcw** [comment]
Thanks for review. Changes made as suggested.

**Comment by P-Copley** [comment]
Looking great @almcw 👍

**Pull request merged**